Capstone/Assets/Game/Scripts/Enemy.cs

```diff
@@ -92,12 +95,16 @@ public virtual void Start()
             death = new EnemyFSMState_DEATH(this);
             wander = new EnemyFSMState_WANDER(this);
             attack = new EnemyFSMState_ATTACK(sword, this);
+            stun = new EnemyFSMState_STUN(sword, this, player);
 
             // Adds all of the possible transitions
             FSM.AddTransition(seek, wander, IsIdle());
             FSM.AddTransition(wander, seek, IsClose());
             FSM.AddTransition(seek, attack, CanAttack());
             FSM.AddTransition(attack, seek, DoneAttacking());
+            FSM.AddTransition(attack, stun, Stunned());
+            FSM.AddTransition(stun, seek, KeepAttacking());
```

Comment:
Should we also have a transition from Stun to Wander in case the player goes out of the effective range while the stun is happening?